### ISSUE TITLE
[ENHANCEMENT]: Support for Zone2 and Buffer tank

### ISSUE THREAD

**Issue by geduxas** [edterbak/HeishaMoNR#132]
**Is your feature request related to a problem or use case? Please describe.**
I am using heatpump with 2 zones of heat. So there is no posibility to control second zone for now. Also i have installed buffer which have additional control for it.

**Describe the solution you'd like**
Would like to have ability to controll Zone2 in same way as Zone1, i think its stright copy of Zone1 element's

**Describe alternatives you've considered**
For now it's controlled manualy.

**Comment by edterbak** [comment]
Hi geduxas,

Thank you for your request.

Can you help me understanding the request fully.
You have 2 heating zones. 
Do you only require Setpoint(s) to be set?
Or do you want additional functionality like the RTC function for Zone 2?

I am unfamiliar with zone2 and buffer setups. Can you draw the picture or explain how the pump is controlling them?
Example, if both zone1 and zone2 are requesting heat, what happens? Can both be fed warm water at the same time, or is there a value which sets this?
Buffer is for zone2 right? and has nothing to do with DHW tank right?

**Comment by edterbak** [comment]
I see there are commands to set the temperatures for zone 1 and 2. I currently only use zone 1 indeed.

![image](https://github.com/edterbak/NodeRed_Heishamon_control/assets/3155621/70f2a309-9389-42dd-976d-8c77a74a0b3d)

In next release, would it be helpfull to offer Zone 2 controls for the temperature in the Settings tab?

**Comment by geduxas** [comment]
Yes, RTC support could be best start. And for other function's i think it will require hard work :)

To run 2 zones heatpump requires upgrade for optional PCB. Both zones can be controlled by heatpump, or one of them as directed. Buffer tank also is optional.

Both zones have separated supply water temperature sensors,  and pump controlled circulating pump (on/off only) and if it's controlled zone it will have it's own mix valve with pump controlled actuator. 

With buffer tank installed heatpump setpoint will be highest temperature of zone +2° so if heat request for Zone 1 will be 36°, Zone 2 - 28,

**Comment by edterbak** [comment]
Hi 

I have just added a little thing in the dashboard whch will help with 2 zones. This is a very small thing, but I wanted to let you know anyway. It is currently on my local version, which is post v23.10 beta8. So the next release (beta9 you should see it. 

![image](https://github.com/edterbak/NodeRed_Heishamon_control/assets/3155621/d8e2ea15-c084-461b-b7d9-c2617ba3f6fb)

**Comment by geduxas** [comment]
I am glad that things are going :) i tried to accomplish by myself, but i am little know node-red, so first of all just renamed all rtc values to rtc_z1 but i think it's little bit wrong :) so thanks!

Now i managed to move my main heating zone from zone2 to zone1 to test all node-red, and i was shocked by possibilities available! 3y using heishamon and just now discovered it :D i think it should be better documented to heishamon itself about possibilities it has.

**Comment by edterbak** [comment]
😄 

Nice

I just noticed you forked and did some work in it ... sweet. 
I briefly had a look at the diff. This helps me for the approach on how to do the z1 and z2 option.

I will already start using the z1 name of variables and setpoints graduatly when I come accross them. 
Slow process, but it is starting now.... little baby steps.

**Comment by edterbak** [comment]
I have changed the stored variable names like this now. (will be part of next beta 15)

![image](https://github.com/edterbak/NodeRed_Heishamon_control/assets/3155621/cba12d3d-294a-4bbd-9daa-c4691e1a8b62)

**Comment by nofeardh** [comment]
Hi,

I saw an debate about including Z2 on Tweakers portal, but unfortunately I cannot reply there.

I'm using 2 zones - one for radiators and one for underfloor heating. From your flow I'm using WAR+RTC (for zone1) and custom flows to control Z2 pump and mixing valve based on Heisha Optional PCB emulation. 

For zone2 it would be nice to have at least WAR and then option to use calculated value inside my personal NodeRed flows.

**Comment by geduxas** [comment]
@edterbak you have done great thing, you have made z1 heating, so it's just need to be replocated for z2, and maybe some dynamic flows by enabling/disabling zones, i am talking about compensation curves, logic is just same. Where is difference, just main pump target temperature set point. You have target set point per zone, and main set point which is highest zone target set point.. i think you're using this in soft start function.. so it will be complicated.. also by enablink buffer tank, you get another temperature sensor. Which you set target delta, if i remember correctly grom 1 to 15, it will add delta amount to highest zone target temperature, and this becomes pump's main target temperature.. 

So for example, z1 target temperature +35°, z2 target - +42°, and you keep bufger tank by 3k, so your main pump target temperature will be +45°. Z1 PID value will go from 100 to 0 by closing it's mixing valve to reduce it's flow temperature to ,+35°, and zone2 also in some time PID 2 will go down tu reduce it's supply temperature to ,42.. and in buffer tank it will try to keep temperature to 45, so it becames main target temperature by which it will reduce compressor frequency..

**Comment by edterbak** [comment]
Indeed. 
This part is for me.. with no zone 2 and no buffer.. very difficult to see 'how' to set things. 
The lack of this understanding... FULLY undertanding it... is keeping me from building it.

It looks to me like I do not need to 'worry' about Z1/Z2 PID things. That is probably not someting I need to use in the flow. The values in Z1/2 PID change automatically, correct?


But, first important question:
Is it **always** the case that when someone uses Zone1+2 (without buffer tank), the highest target will be dominant?

2nd question:
In case of Zone 1+2 with buffer tank, is the highest Z Target also dominant?

Some info about your setup
Are you using (TOP76) 0/1 ?        HeatingMode: DIRECT or CompensationCurve mode?
Are you using (TOP111+112) 0/1/2/3 ? (Water, Room thermostat (Int), Room thermostat (Ext) or Thermistor)

**Comment by geduxas** [comment]
PID value could be indicator to how much heat energy zone requesting.

For my setup i using CompensationCurve, with Room thermostat (Ext), because i went this way for RTC kind of setup..

**Comment by edterbak** [comment]
@geduxas 
Could you please give me more information from heishamon.
At a HEAT moment, can you give me an example of all the values in Heishamon. 
Just copy the entire text table as text here in a msg. If you are able to show more than one scenario, like heat z1 dominant, heat z2 dominant, with(out) buffer, etc. That would be really helpfull.
Same question for @nofeardh 

😃

**Comment by geduxas** [comment]
Buffer tank Off,Z1 Dominant

| Topic | Name | Value | Description|
| --- | --- | --- | --- |
| TOP0 | Heatpump_State | 1 | On|
| TOP1 | Pump_Flow | 19.47 | l/min|
| TOP2 | Force_DHW_State | 0 | Disabled|
| TOP3 | Quiet_Mode_Schedule | 0 | Disabled|
| TOP4 | Operating_Mode_State | 4 | Heat+DHW|
| TOP5 | Main_Inlet_Temp | 27.25 | °C|
| TOP6 | Main_Outlet_Temp | 34.25 | °C|
| TOP7 | Main_Target_Temp | 36 | °C|
| TOP8 | Compressor_Freq | 43 | Hz|
| TOP9 | DHW_Target_Temp | 50 | °C|
| TOP10 | DHW_Temp | 50 | °C|
| TOP11 | Operations_Hours | 19518 | hours|
| TOP12 | Operations_Counter | 20615 | count|
| TOP13 | Main_Schedule_State | 0 | Disabled|
| TOP14 | Outside_Temp | 13 | °C|
| TOP15 | Heat_Power_Production | 9400 | Watt|
| TOP16 | Heat_Power_Consumption | 2200 | Watt|
| TOP17 | Powerful_Mode_Time | 0 | Off|
| TOP18 | Quiet_Mode_Level | 0 | Off|
| TOP19 | Holiday_Mode_State | 0 | Off|
| TOP20 | ThreeWay_Valve_State | 0 | Room|
| TOP21 | Outside_Pipe_Temp | 4 | °C|
| TOP22 | DHW_Heat_Delta | -10 | K|
| TOP23 | Heat_Delta | 5 | K|
| TOP24 | Cool_Delta | 5 | K|
| TOP25 | DHW_Holiday_Shift_Temp | -15 | K|
| TOP26 | Defrosting_State | 0 | Disabled|
| TOP27 | Z1_Heat_Request_Temp | 0 | °C|
| TOP28 | Z1_Cool_Request_Temp | 0 | °C|
| TOP29 | Z1_Heat_Curve_Target_High_Temp | 45 | °C|
| TOP30 | Z1_Heat_Curve_Target_Low_Temp | 35 | °C|
| TOP31 | Z1_Heat_Curve_Outside_High_Temp | 15 | °C|
| TOP32 | Z1_Heat_Curve_Outside_Low_Temp | -5 | °C|
| TOP33 | Room_Thermostat_Temp | 28 | °C|
| TOP34 | Z2_Heat_Request_Temp | 0 | °C|
| TOP35 | Z2_Cool_Request_Temp | 0 | °C|
| TOP36 | Z1_Water_Temp | 28 | °C|
| TOP37 | Z2_Water_Temp | 28 | °C|
| TOP38 | Cool_Power_Production | 0 | Watt|
| TOP39 | Cool_Power_Consumption | 0 | Watt|
| TOP40 | DHW_Power_Production | 0 | Watt|
| TOP41 | DHW_Power_Consumption | 0 | Watt|
| TOP42 | Z1_Water_Target_Temp | 36 | °C|
| TOP43 | Z2_Water_Target_Temp | 28 | °C|
| TOP44 | Error | No error | Error|
| TOP45 | Room_Holiday_Shift_Temp | -15 | K|
| TOP46 | Buffer_Temp | 26 | °C|
| TOP47 | Solar_Temp | -78 | °C|
| TOP48 | Pool_Temp | -78 | °C|
| TOP49 | Main_Hex_Outlet_Temp | 34 | °C|
| TOP50 | Discharge_Temp | 36 | °C|
| TOP51 | Inside_Pipe_Temp | 34 | °C|
| TOP52 | Defrost_Temp | 9 | °C|
| TOP53 | Eva_Outlet_Temp | 7 | °C|
| TOP54 | Bypass_Outlet_Temp | 4 | °C|
| TOP55 | Ipm_Temp | 20 | °C|
| TOP56 | Z1_Temp | 28 | °C|
| TOP57 | Z2_Temp | 28 | °C|
| TOP58 | DHW_Heater_State | 1 | Free|
| TOP59 | Room_Heater_State | 0 | Blocked|
| TOP60 | Internal_Heater_State | 0 | Inactive|
| TOP61 | External_Heater_State | 0 | Inactive|
| TOP62 | Fan1_Motor_Speed | 500 | r/min|
| TOP63 | Fan2_Motor_Speed | 540 | r/min|
| TOP64 | High_Pressure | 21.4 | Kgf/cm2|
| TOP65 | Pump_Speed | 1600 | r/min|
| TOP66 | Low_Pressure | 0 | Kgf/cm2|
| TOP67 | Compressor_Current | 3.4 | Ampere|
| TOP68 | Force_Heater_State | 0 | Inactive|
| TOP69 | Sterilization_State | 0 | Inactive|
| TOP70 | Sterilization_Temp | 62 | °C|
| TOP71 | Sterilization_Max_Time | 20 | Minutes|
| TOP72 | Z1_Cool_Curve_Target_High_Temp | 16 | °C|
| TOP73 | Z1_Cool_Curve_Target_Low_Temp | 10 | °C|
| TOP74 | Z1_Cool_Curve_Outside_High_Temp | 30 | °C|
| TOP75 | Z1_Cool_Curve_Outside_Low_Temp | 20 | °C|
| TOP76 | Heating_Mode | 0 | Comp. Curve|
| TOP77 | Heating_Off_Outdoor_Temp | 17 | °C|
| TOP78 | Heater_On_Outdoor_Temp | -15 | °C|
| TOP79 | Heat_To_Cool_Temp | 15 | °C|
| TOP80 | Cool_To_Heat_Temp | 10 | °C|
| TOP81 | Cooling_Mode | 1 | Direct|
| TOP82 | Z2_Heat_Curve_Target_High_Temp | 35 | °C|
| TOP83 | Z2_Heat_Curve_Target_Low_Temp | 27 | °C|
| TOP84 | Z2_Heat_Curve_Outside_High_Temp | 15 | °C|
| TOP85 | Z2_Heat_Curve_Outside_Low_Temp | -5 | °C|
| TOP86 | Z2_Cool_Curve_Target_High_Temp | 10 | °C|
| TOP87 | Z2_Cool_Curve_Target_Low_Temp | 10 | °C|
| TOP88 | Z2_Cool_Curve_Outside_High_Temp | 30 | °C|
| TOP89 | Z2_Cool_Curve_Outside_Low_Temp | 20 | °C|
| TOP90 | Room_Heater_Operations_Hours | 6 | hours|
| TOP91 | DHW_Heater_Operations_Hours | 60 | hours|
| TOP92 | Heat_Pump_Model | E2 CF 0C 74 09 12 D0 0D 95 05 | Model|
| TOP93 | Pump_Duty | 80 | Duty|
| TOP94 | Zones_State | 2 | Zone1 and zone2 active|
| TOP95 | Max_Pump_Duty | 131 | Duty|
| TOP96 | Heater_Delay_Time | -1 | Minutes|
| TOP97 | Heater_Start_Delta | -128 | K|
| TOP98 | Heater_S| TOP_Delta | -128 | K|
| TOP99 | Buffer_Installed | 0 | Disabled|
| TOP100 | DHW_Installed | 1 | Enabled|
| TOP101 | Solar_Mode | 0 | Disabled|
| TOP102 | Solar_On_Delta | 10 | K|
| TOP103 | Solar_Off_Delta | 5 | K|
| TOP104 | Solar_Frost_Protection | 5 | °C|
| TOP105 | Solar_High_Limit | 80 | °C|
| TOP106 | Pump_Flowrate_Mode | -1 | unknown|
| TOP107 | Liquid_Type | 0 | Water|
| TOP108 | Alt_External_Sensor | 0 | Disabled|
| TOP109 | Anti_Freeze_Mode | 1 | Enabled|
| TOP110 | Optional_PCB | 1 | Enabled|
| TOP111 | Z1_Sensor_Settings | 1 | External Thermostat|
| TOP112 | Z2_Sensor_Settings | 1 | External Thermostat|
| TOP113 | Buffer_Tank_Delta | 0 | K|
| TOP114 | External_Pad_Heater | 0 | Disabled|
| TOP115 | Water_Pressure | -0.02 | Bar|
| TOP116 | Second_Inlet_Temp | -128 | °C|
| TOP117 | Economizer_Outlet_Temp | -128 | °C|
| TOP118 | Second_Room_Thermostat_Temp | -128 | °C|
| TOP119 | External_Control | 0 | Disabled|
| TOP120 | External_Heat_Cool_Control | 0 | Disabled|
| TOP121 | External_Error_Signal | 0 | Disabled|
| TOP122 | External_Compressor_Control | 0 | Disabled|
| TOP123 | Z2_Pump_State | 1 | On|
| TOP124 | Z1_Pump_State | 1 | On|
| TOP125 | TwoWay_Valve_State | 1 | Heat|
| TOP126 | ThreeWay_Valve_State2 | 0 | Room|
| TOP127 | Z1_Valve_PID | 100 | %|
| TOP128 | Z2_Valve_PID | 100 | %|
| TOP129 | Bivalent_Control | 0 | Disabled|
| TOP130 | Bivalent_Mode | 0 | Alternative|
| TOP131 | Bivalent_Start_Temp | -5 | °C|
| TOP132 | Bivalent_Advanced_Heat | 0 | Disabled|
| TOP133 | Bivalent_Advanced_DHW | 0 | Disabled|
| TOP134 | Bivalent_Advanced_Start_Temp | -8 | °C|
| TOP135 | Bivalent_Advanced_S| TOP_Temp | -2 | °C|
| TOP136 | Bivalent_Advanced_Start_Delay | 30 | Minutes|
| TOP137 | Bivalent_Advanced_S| TOP_Delay | 30 | Minutes|
| TOP138 | Bivalent_Advanced_DHW_Delay | 30 | Minutes|
| TOP139 | EEV_Valve | 74 | count|
| TOP140 | ByPass_Valve | 0 | count|



Buffer tank Off,Z2 Dominant

| Topic | Name | Value | Description|
| --- | --- | --- | --- |
| TOP0 | Heatpump_State | 1 | On|
| TOP1 | Pump_Flow | 34.86 | l/min|
| TOP2 | Force_DHW_State | 0 | Disabled|
| TOP3 | Quiet_Mode_Schedule | 0 | Disabled|
| TOP4 | Operating_Mode_State | 4 | Heat+DHW|
| TOP5 | Main_Inlet_Temp | 26.75 | °C|
| TOP6 | Main_Outlet_Temp | 28.75 | °C|
| TOP7 | Main_Target_Temp | 36 | °C|
| TOP8 | Compressor_Freq | 21 | Hz|
| TOP9 | DHW_Target_Temp | 50 | °C|
| TOP10 | DHW_Temp | 49 | °C|
| TOP11 | Operations_Hours | 19518 | hours|
| TOP12 | Operations_Counter | 20616 | count|
| TOP13 | Main_Schedule_State | 0 | Disabled|
| TOP14 | Outside_Temp | 12 | °C|
| TOP15 | Heat_Power_Production | 4400 | Watt|
| TOP16 | Heat_Power_Consumption | 800 | Watt|
| TOP17 | Powerful_Mode_Time | 0 | Off|
| TOP18 | Quiet_Mode_Level | 0 | Off|
| TOP19 | Holiday_Mode_State | 0 | Off|
| TOP20 | ThreeWay_Valve_State | 0 | Room|
| TOP21 | Outside_Pipe_Temp | 7 | °C|
| TOP22 | DHW_Heat_Delta | -10 | K|
| TOP23 | Heat_Delta | 5 | K|
| TOP24 | Cool_Delta | 5 | K|
| TOP25 | DHW_Holiday_Shift_Temp | -15 | K|
| TOP26 | Defrosting_State | 0 | Disabled|
| TOP27 | Z1_Heat_Request_Temp | 0 | °C|
| TOP28 | Z1_Cool_Request_Temp | 0 | °C|
| TOP29 | Z1_Heat_Curve_Target_High_Temp | 35 | °C|
| TOP30 | Z1_Heat_Curve_Target_Low_Temp | 27 | °C|
| TOP31 | Z1_Heat_Curve_Outside_High_Temp | 15 | °C|
| TOP32 | Z1_Heat_Curve_Outside_Low_Temp | -5 | °C|
| TOP33 | Room_Thermostat_Temp | 28 | °C|
| TOP34 | Z2_Heat_Request_Temp | 0 | °C|
| TOP35 | Z2_Cool_Request_Temp | 0 | °C|
| TOP36 | Z1_Water_Temp | 27 | °C|
| TOP37 | Z2_Water_Temp | 27 | °C|
| TOP38 | Cool_Power_Production | 0 | Watt|
| TOP39 | Cool_Power_Consumption | 0 | Watt|
| TOP40 | DHW_Power_Production | 0 | Watt|
| TOP41 | DHW_Power_Consumption | 0 | Watt|
| TOP42 | Z1_Water_Target_Temp | 28 | °C|
| TOP43 | Z2_Water_Target_Temp | 36 | °C|
| TOP44 | Error | No error | Error|
| TOP45 | Room_Holiday_Shift_Temp | -15 | K|
| TOP46 | Buffer_Temp | 27 | °C|
| TOP47 | Solar_Temp | -78 | °C|
| TOP48 | Pool_Temp | -78 | °C|
| TOP49 | Main_Hex_Outlet_Temp | 28 | °C|
| TOP50 | Discharge_Temp | 34 | °C|
| TOP51 | Inside_Pipe_Temp | 29 | °C|
| TOP52 | Defrost_Temp | 8 | °C|
| TOP53 | Eva_Outlet_Temp | 7 | °C|
| TOP54 | Bypass_Outlet_Temp | 9 | °C|
| TOP55 | Ipm_Temp | 18 | °C|
| TOP56 | Z1_Temp | 27 | °C|
| TOP57 | Z2_Temp | 27 | °C|
| TOP58 | DHW_Heater_State | 1 | Free|
| TOP59 | Room_Heater_State | 0 | Blocked|
| TOP60 | Internal_Heater_State | 0 | Inactive|
| TOP61 | External_Heater_State | 0 | Inactive|
| TOP62 | Fan1_Motor_Speed | 350 | r/min|
| TOP63 | Fan2_Motor_Speed | 390 | r/min|
| TOP64 | High_Pressure | 18.2 | Kgf/cm2|
| TOP65 | Pump_Speed | 2700 | r/min|
| TOP66 | Low_Pressure | 0 | Kgf/cm2|
| TOP67 | Compressor_Current | 1.4 | Ampere|
| TOP68 | Force_Heater_State | 0 | Inactive|
| TOP69 | Sterilization_State | 0 | Inactive|
| TOP70 | Sterilization_Temp | 62 | °C|
| TOP71 | Sterilization_Max_Time | 20 | Minutes|
| TOP72 | Z1_Cool_Curve_Target_High_Temp | 16 | °C|
| TOP73 | Z1_Cool_Curve_Target_Low_Temp | 10 | °C|
| TOP74 | Z1_Cool_Curve_Outside_High_Temp | 30 | °C|
| TOP75 | Z1_Cool_Curve_Outside_Low_Temp | 20 | °C|
| TOP76 | Heating_Mode | 0 | Comp. Curve|
| TOP77 | Heating_Off_Outdoor_Temp | 17 | °C|
| TOP78 | Heater_On_Outdoor_Temp | -15 | °C|
| TOP79 | Heat_To_Cool_Temp | 15 | °C|
| TOP80 | Cool_To_Heat_Temp | 10 | °C|
| TOP81 | Cooling_Mode | 1 | Direct|
| TOP82 | Z2_Heat_Curve_Target_High_Temp | 45 | °C|
| TOP83 | Z2_Heat_Curve_Target_Low_Temp | 35 | °C|
| TOP84 | Z2_Heat_Curve_Outside_High_Temp | 15 | °C|
| TOP85 | Z2_Heat_Curve_Outside_Low_Temp | -5 | °C|
| TOP86 | Z2_Cool_Curve_Target_High_Temp | 10 | °C|
| TOP87 | Z2_Cool_Curve_Target_Low_Temp | 10 | °C|
| TOP88 | Z2_Cool_Curve_Outside_High_Temp | 30 | °C|
| TOP89 | Z2_Cool_Curve_Outside_Low_Temp | 20 | °C|
| TOP90 | Room_Heater_Operations_Hours | 6 | hours|
| TOP91 | DHW_Heater_Operations_Hours | 60 | hours|
| TOP92 | Heat_Pump_Model | E2 CF 0C 74 09 12 D0 0D 95 05 | Model|
| TOP93 | Pump_Duty | 108 | Duty|
| TOP94 | Zones_State | 2 | Zone1 and zone2 active|
| TOP95 | Max_Pump_Duty | 131 | Duty|
| TOP96 | Heater_Delay_Time | -1 | Minutes|
| TOP97 | Heater_Start_Delta | -128 | K|
| TOP98 | Heater_S| TOP_Delta | -128 | K|
| TOP99 | Buffer_Installed | 0 | Disabled|
| TOP100 | DHW_Installed | 1 | Enabled|
| TOP101 | Solar_Mode | 0 | Disabled|
| TOP102 | Solar_On_Delta | 10 | K|
| TOP103 | Solar_Off_Delta | 5 | K|
| TOP104 | Solar_Frost_Protection | 5 | °C|
| TOP105 | Solar_High_Limit | 80 | °C|
| TOP106 | Pump_Flowrate_Mode | -1 | unknown|
| TOP107 | Liquid_Type | 0 | Water|
| TOP108 | Alt_External_Sensor | 0 | Disabled|
| TOP109 | Anti_Freeze_Mode | 1 | Enabled|
| TOP110 | Optional_PCB | 1 | Enabled|
| TOP111 | Z1_Sensor_Settings | 1 | External Thermostat|
| TOP112 | Z2_Sensor_Settings | 1 | External Thermostat|
| TOP113 | Buffer_Tank_Delta | 3 | K|
| TOP114 | External_Pad_Heater | 0 | Disabled|
| TOP115 | Water_Pressure | -0.02 | Bar|
| TOP116 | Second_Inlet_Temp | -128 | °C|
| TOP117 | Economizer_Outlet_Temp | -128 | °C|
| TOP118 | Second_Room_Thermostat_Temp | -128 | °C|
| TOP119 | External_Control | 0 | Disabled|
| TOP120 | External_Heat_Cool_Control | 0 | Disabled|
| TOP121 | External_Error_Signal | 0 | Disabled|
| TOP122 | External_Compressor_Control | 0 | Disabled|
| TOP123 | Z2_Pump_State | 1 | On|
| TOP124 | Z1_Pump_State | 1 | On|
| TOP125 | TwoWay_Valve_State | 1 | Heat|
| TOP126 | ThreeWay_Valve_State2 | 0 | Room|
| TOP127 | Z1_Valve_PID | 100 | %|
| TOP128 | Z2_Valve_PID | 100 | %|
| TOP129 | Bivalent_Control | 0 | Disabled|
| TOP130 | Bivalent_Mode | 0 | Alternative|
| TOP131 | Bivalent_Start_Temp | -5 | °C|
| TOP132 | Bivalent_Advanced_Heat | 0 | Disabled|
| TOP133 | Bivalent_Advanced_DHW | 0 | Disabled|
| TOP134 | Bivalent_Advanced_Start_Temp | -8 | °C|
| TOP135 | Bivalent_Advanced_S| TOP_Temp | -2 | °C|
| TOP136 | Bivalent_Advanced_Start_Delay | 30 | Minutes|
| TOP137 | Bivalent_Advanced_S| TOP_Delay | 30 | Minutes|
| TOP138 | Bivalent_Advanced_DHW_Delay | 30 | Minutes|
| TOP139 | EEV_Valve | 74 | count|
| TOP140 | ByPass_Valve | 0 | count|



Buffer Tank On Delta3, Z1 Dominant

| Topic | Name | Value | Description|
| --- | --- | --- | --- |
| TOP0 | Heatpump_State | 1 | On|
| TOP1 | Pump_Flow | 34.86 | l/min|
| TOP2 | Force_DHW_State | 0 | Disabled|
| TOP3 | Quiet_Mode_Schedule | 0 | Disabled|
| TOP4 | Operating_Mode_State | 4 | Heat+DHW|
| TOP5 | Main_Inlet_Temp | 28.25 | °C|
| TOP6 | Main_Outlet_Temp | 32.25 | °C|
| TOP7 | Main_Target_Temp | 41 | °C|
| TOP8 | Compressor_Freq | 0 | Hz|
| TOP9 | DHW_Target_Temp | 50 | °C|
| TOP10 | DHW_Temp | 49 | °C|
| TOP11 | Operations_Hours | 19518 | hours|
| TOP12 | Operations_Counter | 20615 | count|
| TOP13 | Main_Schedule_State | 0 | Disabled|
| TOP14 | Outside_Temp | 13 | °C|
| TOP15 | Heat_Power_Production | 0 | Watt|
| TOP16 | Heat_Power_Consumption | 200 | Watt|
| TOP17 | Powerful_Mode_Time | 0 | Off|
| TOP18 | Quiet_Mode_Level | 0 | Off|
| TOP19 | Holiday_Mode_State | 0 | Off|
| TOP20 | ThreeWay_Valve_State | 0 | Room|
| TOP21 | Outside_Pipe_Temp | 5 | °C|
| TOP22 | DHW_Heat_Delta | -10 | K|
| TOP23 | Heat_Delta | 5 | K|
| TOP24 | Cool_Delta | 5 | K|
| TOP25 | DHW_Holiday_Shift_Temp | -15 | K|
| TOP26 | Defrosting_State | 0 | Disabled|
| TOP27 | Z1_Heat_Request_Temp | 0 | °C|
| TOP28 | Z1_Cool_Request_Temp | 0 | °C|
| TOP29 | Z1_Heat_Curve_Target_High_Temp | 45 | °C|
| TOP30 | Z1_Heat_Curve_Target_Low_Temp | 35 | °C|
| TOP31 | Z1_Heat_Curve_Outside_High_Temp | 15 | °C|
| TOP32 | Z1_Heat_Curve_Outside_Low_Temp | -5 | °C|
| TOP33 | Room_Thermostat_Temp | 28 | °C|
| TOP34 | Z2_Heat_Request_Temp | 0 | °C|
| TOP35 | Z2_Cool_Request_Temp | 0 | °C|
| TOP36 | Z1_Water_Temp | 29 | °C|
| TOP37 | Z2_Water_Temp | 29 | °C|
| TOP38 | Cool_Power_Production | 0 | Watt|
| TOP39 | Cool_Power_Consumption | 0 | Watt|
| TOP40 | DHW_Power_Production | 0 | Watt|
| TOP41 | DHW_Power_Consumption | 0 | Watt|
| TOP42 | Z1_Water_Target_Temp | 36 | °C|
| TOP43 | Z2_Water_Target_Temp | 28 | °C|
| TOP44 | Error | No error | Error|
| TOP45 | Room_Holiday_Shift_Temp | -15 | K|
| TOP46 | Buffer_Temp | 26 | °C|
| TOP47 | Solar_Temp | -78 | °C|
| TOP48 | Pool_Temp | -78 | °C|
| TOP49 | Main_Hex_Outlet_Temp | 31 | °C|
| TOP50 | Discharge_Temp | 38 | °C|
| TOP51 | Inside_Pipe_Temp | 29 | °C|
| TOP52 | Defrost_Temp | 9 | °C|
| TOP53 | Eva_Outlet_Temp | 5 | °C|
| TOP54 | Bypass_Outlet_Temp | 7 | °C|
| TOP55 | Ipm_Temp | 20 | °C|
| TOP56 | Z1_Temp | 29 | °C|
| TOP57 | Z2_Temp | 29 | °C|
| TOP58 | DHW_Heater_State | 1 | Free|
| TOP59 | Room_Heater_State | 0 | Blocked|
| TOP60 | Internal_Heater_State | 0 | Inactive|
| TOP61 | External_Heater_State | 0 | Inactive|
| TOP62 | Fan1_Motor_Speed | 580 | r/min|
| TOP63 | Fan2_Motor_Speed | 620 | r/min|
| TOP64 | High_Pressure | 15.6 | Kgf/cm2|
| TOP65 | Pump_Speed | 2650 | r/min|
| TOP66 | Low_Pressure | 0 | Kgf/cm2|
| TOP67 | Compressor_Current | 0.6 | Ampere|
| TOP68 | Force_Heater_State | 0 | Inactive|
| TOP69 | Sterilization_State | 0 | Inactive|
| TOP70 | Sterilization_Temp | 62 | °C|
| TOP71 | Sterilization_Max_Time | 20 | Minutes|
| TOP72 | Z1_Cool_Curve_Target_High_Temp | 16 | °C|
| TOP73 | Z1_Cool_Curve_Target_Low_Temp | 10 | °C|
| TOP74 | Z1_Cool_Curve_Outside_High_Temp | 30 | °C|
| TOP75 | Z1_Cool_Curve_Outside_Low_Temp | 20 | °C|
| TOP76 | Heating_Mode | 0 | Comp. Curve|
| TOP77 | Heating_Off_Outdoor_Temp | 17 | °C|
| TOP78 | Heater_On_Outdoor_Temp | -15 | °C|
| TOP79 | Heat_To_Cool_Temp | 15 | °C|
| TOP80 | Cool_To_Heat_Temp | 10 | °C|
| TOP81 | Cooling_Mode | 1 | Direct|
| TOP82 | Z2_Heat_Curve_Target_High_Temp | 35 | °C|
| TOP83 | Z2_Heat_Curve_Target_Low_Temp | 27 | °C|
| TOP84 | Z2_Heat_Curve_Outside_High_Temp | 15 | °C|
| TOP85 | Z2_Heat_Curve_Outside_Low_Temp | -5 | °C|
| TOP86 | Z2_Cool_Curve_Target_High_Temp | 10 | °C|
| TOP87 | Z2_Cool_Curve_Target_Low_Temp | 10 | °C|
| TOP88 | Z2_Cool_Curve_Outside_High_Temp | 30 | °C|
| TOP89 | Z2_Cool_Curve_Outside_Low_Temp | 20 | °C|
| TOP90 | Room_Heater_Operations_Hours | 6 | hours|
| TOP91 | DHW_Heater_Operations_Hours | 60 | hours|
| TOP92 | Heat_Pump_Model | E2 CF 0C 74 09 12 D0 0D 95 05 | Model|
| TOP93 | Pump_Duty | 108 | Duty|
| TOP94 | Zones_State | 2 | Zone1 and zone2 active|
| TOP95 | Max_Pump_Duty | 131 | Duty|
| TOP96 | Heater_Delay_Time | -1 | Minutes|
| TOP97 | Heater_Start_Delta | -128 | K|
| TOP98 | Heater_S| TOP_Delta | -128 | K|
| TOP99 | Buffer_Installed | 1 | Enabled|
| TOP100 | DHW_Installed | 1 | Enabled|
| TOP101 | Solar_Mode | 0 | Disabled|
| TOP102 | Solar_On_Delta | 10 | K|
| TOP103 | Solar_Off_Delta | 5 | K|
| TOP104 | Solar_Frost_Protection | 5 | °C|
| TOP105 | Solar_High_Limit | 80 | °C|
| TOP106 | Pump_Flowrate_Mode | -1 | unknown|
| TOP107 | Liquid_Type | 0 | Water|
| TOP108 | Alt_External_Sensor | 0 | Disabled|
| TOP109 | Anti_Freeze_Mode | 1 | Enabled|
| TOP110 | Optional_PCB | 1 | Enabled|
| TOP111 | Z1_Sensor_Settings | 1 | External Thermostat|
| TOP112 | Z2_Sensor_Settings | 1 | External Thermostat|
| TOP113 | Buffer_Tank_Delta | 3 | K|
| TOP114 | External_Pad_Heater | 0 | Disabled|
| TOP115 | Water_Pressure | -0.02 | Bar|
| TOP116 | Second_Inlet_Temp | -128 | °C|
| TOP117 | Economizer_Outlet_Temp | -128 | °C|
| TOP118 | Second_Room_Thermostat_Temp | -128 | °C|
| TOP119 | External_Control | 0 | Disabled|
| TOP120 | External_Heat_Cool_Control | 0 | Disabled|
| TOP121 | External_Error_Signal | 0 | Disabled|
| TOP122 | External_Compressor_Control | 0 | Disabled|
| TOP123 | Z2_Pump_State | 1 | On|
| TOP124 | Z1_Pump_State | 1 | On|
| TOP125 | TwoWay_Valve_State | 1 | Heat|
| TOP126 | ThreeWay_Valve_State2 | 0 | Room|
| TOP127 | Z1_Valve_PID | 100 | %|
| TOP128 | Z2_Valve_PID | 99 | %|
| TOP129 | Bivalent_Control | 0 | Disabled|
| TOP130 | Bivalent_Mode | 0 | Alternative|
| TOP131 | Bivalent_Start_Temp | -5 | °C|
| TOP132 | Bivalent_Advanced_Heat | 0 | Disabled|
| TOP133 | Bivalent_Advanced_DHW | 0 | Disabled|
| TOP134 | Bivalent_Advanced_Start_Temp | -8 | °C|
| TOP135 | Bivalent_Advanced_S| TOP_Temp | -2 | °C|
| TOP136 | Bivalent_Advanced_Start_Delay | 30 | Minutes|
| TOP137 | Bivalent_Advanced_S| TOP_Delay | 30 | Minutes|
| TOP138 | Bivalent_Advanced_DHW_Delay | 30 | Minutes|
| TOP139 | EEV_Valve | 108 | count|
| TOP140 | ByPass_Valve | 120 | count|



Buffer Tank On Delta3, Z2 Dominant

| Topic | Name | Value | Description|
| --- | --- | --- | --- |
| TOP0 | Heatpump_State | 1 | On|
| TOP1 | Pump_Flow | 35.22 | l/min|
| TOP2 | Force_DHW_State | 0 | Disabled|
| TOP3 | Quiet_Mode_Schedule | 0 | Disabled|
| TOP4 | Operating_Mode_State | 4 | Heat+DHW|
| TOP5 | Main_Inlet_Temp | 26.5 | °C|
| TOP6 | Main_Outlet_Temp | 27.5 | °C|
| TOP7 | Main_Target_Temp | 41 | °C|
| TOP8 | Compressor_Freq | 19 | Hz|
| TOP9 | DHW_Target_Temp | 50 | °C|
| TOP10 | DHW_Temp | 49 | °C|
| TOP11 | Operations_Hours | 19518 | hours|
| TOP12 | Operations_Counter | 20616 | count|
| TOP13 | Main_Schedule_State | 0 | Disabled|
| TOP14 | Outside_Temp | 12 | °C|
| TOP15 | Heat_Power_Production | 2600 | Watt|
| TOP16 | Heat_Power_Consumption | 600 | Watt|
| TOP17 | Powerful_Mode_Time | 0 | Off|
| TOP18 | Quiet_Mode_Level | 0 | Off|
| TOP19 | Holiday_Mode_State | 0 | Off|
| TOP20 | ThreeWay_Valve_State | 0 | Room|
| TOP21 | Outside_Pipe_Temp | 11 | °C|
| TOP22 | DHW_Heat_Delta | -10 | K|
| TOP23 | Heat_Delta | 5 | K|
| TOP24 | Cool_Delta | 5 | K|
| TOP25 | DHW_Holiday_Shift_Temp | -15 | K|
| TOP26 | Defrosting_State | 0 | Disabled|
| TOP27 | Z1_Heat_Request_Temp | 0 | °C|
| TOP28 | Z1_Cool_Request_Temp | 0 | °C|
| TOP29 | Z1_Heat_Curve_Target_High_Temp | 35 | °C|
| TOP30 | Z1_Heat_Curve_Target_Low_Temp | 27 | °C|
| TOP31 | Z1_Heat_Curve_Outside_High_Temp | 15 | °C|
| TOP32 | Z1_Heat_Curve_Outside_Low_Temp | -5 | °C|
| TOP33 | Room_Thermostat_Temp | 28 | °C|
| TOP34 | Z2_Heat_Request_Temp | 0 | °C|
| TOP35 | Z2_Cool_Request_Temp | 0 | °C|
| TOP36 | Z1_Water_Temp | 26 | °C|
| TOP37 | Z2_Water_Temp | 26 | °C|
| TOP38 | Cool_Power_Production | 0 | Watt|
| TOP39 | Cool_Power_Consumption | 0 | Watt|
| TOP40 | DHW_Power_Production | 0 | Watt|
| TOP41 | DHW_Power_Consumption | 0 | Watt|
| TOP42 | Z1_Water_Target_Temp | 28 | °C|
| TOP43 | Z2_Water_Target_Temp | 36 | °C|
| TOP44 | Error | No error | Error|
| TOP45 | Room_Holiday_Shift_Temp | -15 | K|
| TOP46 | Buffer_Temp | 27 | °C|
| TOP47 | Solar_Temp | -78 | °C|
| TOP48 | Pool_Temp | -78 | °C|
| TOP49 | Main_Hex_Outlet_Temp | 27 | °C|
| TOP50 | Discharge_Temp | 34 | °C|
| TOP51 | Inside_Pipe_Temp | 28 | °C|
| TOP52 | Defrost_Temp | 11 | °C|
| TOP53 | Eva_Outlet_Temp | 12 | °C|
| TOP54 | Bypass_Outlet_Temp | 16 | °C|
| TOP55 | Ipm_Temp | 17 | °C|
| TOP56 | Z1_Temp | 26 | °C|
| TOP57 | Z2_Temp | 26 | °C|
| TOP58 | DHW_Heater_State | 1 | Free|
| TOP59 | Room_Heater_State | 0 | Blocked|
| TOP60 | Internal_Heater_State | 0 | Inactive|
| TOP61 | External_Heater_State | 0 | Inactive|
| TOP62 | Fan1_Motor_Speed | 350 | r/min|
| TOP63 | Fan2_Motor_Speed | 390 | r/min|
| TOP64 | High_Pressure | 17.8 | Kgf/cm2|
| TOP65 | Pump_Speed | 2700 | r/min|
| TOP66 | Low_Pressure | 0 | Kgf/cm2|
| TOP67 | Compressor_Current | 1.2 | Ampere|
| TOP68 | Force_Heater_State | 0 | Inactive|
| TOP69 | Sterilization_State | 0 | Inactive|
| TOP70 | Sterilization_Temp | 62 | °C|
| TOP71 | Sterilization_Max_Time | 20 | Minutes|
| TOP72 | Z1_Cool_Curve_Target_High_Temp | 16 | °C|
| TOP73 | Z1_Cool_Curve_Target_Low_Temp | 10 | °C|
| TOP74 | Z1_Cool_Curve_Outside_High_Temp | 30 | °C|
| TOP75 | Z1_Cool_Curve_Outside_Low_Temp | 20 | °C|
| TOP76 | Heating_Mode | 0 | Comp. Curve|
| TOP77 | Heating_Off_Outdoor_Temp | 17 | °C|
| TOP78 | Heater_On_Outdoor_Temp | -15 | °C|
| TOP79 | Heat_To_Cool_Temp | 15 | °C|
| TOP80 | Cool_To_Heat_Temp | 10 | °C|
| TOP81 | Cooling_Mode | 1 | Direct|
| TOP82 | Z2_Heat_Curve_Target_High_Temp | 45 | °C|
| TOP83 | Z2_Heat_Curve_Target_Low_Temp | 35 | °C|
| TOP84 | Z2_Heat_Curve_Outside_High_Temp | 15 | °C|
| TOP85 | Z2_Heat_Curve_Outside_Low_Temp | -5 | °C|
| TOP86 | Z2_Cool_Curve_Target_High_Temp | 10 | °C|
| TOP87 | Z2_Cool_Curve_Target_Low_Temp | 10 | °C|
| TOP88 | Z2_Cool_Curve_Outside_High_Temp | 30 | °C|
| TOP89 | Z2_Cool_Curve_Outside_Low_Temp | 20 | °C|
| TOP90 | Room_Heater_Operations_Hours | 6 | hours|
| TOP91 | DHW_Heater_Operations_Hours | 60 | hours|
| TOP92 | Heat_Pump_Model | E2 CF 0C 74 09 12 D0 0D 95 05 | Model|
| TOP93 | Pump_Duty | 108 | Duty|
| TOP94 | Zones_State | 2 | Zone1 and zone2 active|
| TOP95 | Max_Pump_Duty | 131 | Duty|
| TOP96 | Heater_Delay_Time | -1 | Minutes|
| TOP97 | Heater_Start_Delta | -128 | K|
| TOP98 | Heater_S| TOP_Delta | -128 | K|
| TOP99 | Buffer_Installed | 1 | Enabled|
| TOP100 | DHW_Installed | 1 | Enabled|
| TOP101 | Solar_Mode | 0 | Disabled|
| TOP102 | Solar_On_Delta | 10 | K|
| TOP103 | Solar_Off_Delta | 5 | K|
| TOP104 | Solar_Frost_Protection | 5 | °C|
| TOP105 | Solar_High_Limit | 80 | °C|
| TOP106 | Pump_Flowrate_Mode | -1 | unknown|
| TOP107 | Liquid_Type | 0 | Water|
| TOP108 | Alt_External_Sensor | 0 | Disabled|
| TOP109 | Anti_Freeze_Mode | 1 | Enabled|
| TOP110 | Optional_PCB | 1 | Enabled|
| TOP111 | Z1_Sensor_Settings | 1 | External Thermostat|
| TOP112 | Z2_Sensor_Settings | 1 | External Thermostat|
| TOP113 | Buffer_Tank_Delta | 3 | K|
| TOP114 | External_Pad_Heater | 0 | Disabled|
| TOP115 | Water_Pressure | -0.02 | Bar|
| TOP116 | Second_Inlet_Temp | -128 | °C|
| TOP117 | Economizer_Outlet_Temp | -128 | °C|
| TOP118 | Second_Room_Thermostat_Temp | -128 | °C|
| TOP119 | External_Control | 0 | Disabled|
| TOP120 | External_Heat_Cool_Control | 0 | Disabled|
| TOP121 | External_Error_Signal | 0 | Disabled|
| TOP122 | External_Compressor_Control | 0 | Disabled|
| TOP123 | Z2_Pump_State | 1 | On|
| TOP124 | Z1_Pump_State | 1 | On|
| TOP125 | TwoWay_Valve_State | 1 | Heat|
| TOP126 | ThreeWay_Valve_State2 | 0 | Room|
| TOP127 | Z1_Valve_PID | 100 | %|
| TOP128 | Z2_Valve_PID | 100 | %|
| TOP129 | Bivalent_Control | 0 | Disabled|
| TOP130 | Bivalent_Mode | 0 | Alternative|
| TOP131 | Bivalent_Start_Temp | -5 | °C|
| TOP132 | Bivalent_Advanced_Heat | 0 | Disabled|
| TOP133 | Bivalent_Advanced_DHW | 0 | Disabled|
| TOP134 | Bivalent_Advanced_Start_Temp | -8 | °C|
| TOP135 | Bivalent_Advanced_S| TOP_Temp | -2 | °C|
| TOP136 | Bivalent_Advanced_Start_Delay | 30 | Minutes|
| TOP137 | Bivalent_Advanced_S| TOP_Delay | 30 | Minutes|
| TOP138 | Bivalent_Advanced_DHW_Delay | 30 | Minutes|
| TOP139 | EEV_Valve | 70 | count|
| TOP140 | ByPass_Valve | 0 | count|

Buffer Tank On Delta10, Z1 Dominant

| Topic | Name | Value | Description|
| --- | --- | --- | --- |
| TOP0 | Heatpump_State | 1 | On|
| TOP1 | Pump_Flow | 35.22 | l/min|
| TOP2 | Force_DHW_State | 0 | Disabled|
| TOP3 | Quiet_Mode_Schedule | 0 | Disabled|
| TOP4 | Operating_Mode_State | 4 | Heat+DHW|
| TOP5 | Main_Inlet_Temp | 28 | °C|
| TOP6 | Main_Outlet_Temp | 28.25 | °C|
| TOP7 | Main_Target_Temp | 48 | °C|
| TOP8 | Compressor_Freq | 0 | Hz|
| TOP9 | DHW_Target_Temp | 50 | °C|
| TOP10 | DHW_Temp | 49 | °C|
| TOP11 | Operations_Hours | 19518 | hours|
| TOP12 | Operations_Counter | 20615 | count|
| TOP13 | Main_Schedule_State | 0 | Disabled|
| TOP14 | Outside_Temp | 13 | °C|
| TOP15 | Heat_Power_Production | 0 | Watt|
| TOP16 | Heat_Power_Consumption | 200 | Watt|
| TOP17 | Powerful_Mode_Time | 0 | Off|
| TOP18 | Quiet_Mode_Level | 0 | Off|
| TOP19 | Holiday_Mode_State | 0 | Off|
| TOP20 | ThreeWay_Valve_State | 0 | Room|
| TOP21 | Outside_Pipe_Temp | 11 | °C|
| TOP22 | DHW_Heat_Delta | -10 | K|
| TOP23 | Heat_Delta | 5 | K|
| TOP24 | Cool_Delta | 5 | K|
| TOP25 | DHW_Holiday_Shift_Temp | -15 | K|
| TOP26 | Defrosting_State | 0 | Disabled|
| TOP27 | Z1_Heat_Request_Temp | 0 | °C|
| TOP28 | Z1_Cool_Request_Temp | 0 | °C|
| TOP29 | Z1_Heat_Curve_Target_High_Temp | 45 | °C|
| TOP30 | Z1_Heat_Curve_Target_Low_Temp | 35 | °C|
| TOP31 | Z1_Heat_Curve_Outside_High_Temp | 15 | °C|
| TOP32 | Z1_Heat_Curve_Outside_Low_Temp | -5 | °C|
| TOP33 | Room_Thermostat_Temp | 28 | °C|
| TOP34 | Z2_Heat_Request_Temp | 0 | °C|
| TOP35 | Z2_Cool_Request_Temp | 0 | °C|
| TOP36 | Z1_Water_Temp | 27 | °C|
| TOP37 | Z2_Water_Temp | 27 | °C|
| TOP38 | Cool_Power_Production | 0 | Watt|
| TOP39 | Cool_Power_Consumption | 0 | Watt|
| TOP40 | DHW_Power_Production | 0 | Watt|
| TOP41 | DHW_Power_Consumption | 0 | Watt|
| TOP42 | Z1_Water_Target_Temp | 36 | °C|
| TOP43 | Z2_Water_Target_Temp | 28 | °C|
| TOP44 | Error | No error | Error|
| TOP45 | Room_Holiday_Shift_Temp | -15 | K|
| TOP46 | Buffer_Temp | 27 | °C|
| TOP47 | Solar_Temp | -78 | °C|
| TOP48 | Pool_Temp | -78 | °C|
| TOP49 | Main_Hex_Outlet_Temp | 28 | °C|
| TOP50 | Discharge_Temp | 28 | °C|
| TOP51 | Inside_Pipe_Temp | 28 | °C|
| TOP52 | Defrost_Temp | 11 | °C|
| TOP53 | Eva_Outlet_Temp | 12 | °C|
| TOP54 | Bypass_Outlet_Temp | 14 | °C|
| TOP55 | Ipm_Temp | 18 | °C|
| TOP56 | Z1_Temp | 27 | °C|
| TOP57 | Z2_Temp | 27 | °C|
| TOP58 | DHW_Heater_State | 1 | Free|
| TOP59 | Room_Heater_State | 0 | Blocked|
| TOP60 | Internal_Heater_State | 0 | Inactive|
| TOP61 | External_Heater_State | 0 | Inactive|
| TOP62 | Fan1_Motor_Speed | 580 | r/min|
| TOP63 | Fan2_Motor_Speed | 620 | r/min|
| TOP64 | High_Pressure | 11 | Kgf/cm2|
| TOP65 | Pump_Speed | 2700 | r/min|
| TOP66 | Low_Pressure | 0 | Kgf/cm2|
| TOP67 | Compressor_Current | 0.6 | Ampere|
| TOP68 | Force_Heater_State | 0 | Inactive|
| TOP69 | Sterilization_State | 0 | Inactive|
| TOP70 | Sterilization_Temp | 62 | °C|
| TOP71 | Sterilization_Max_Time | 20 | Minutes|
| TOP72 | Z1_Cool_Curve_Target_High_Temp | 16 | °C|
| TOP73 | Z1_Cool_Curve_Target_Low_Temp | 10 | °C|
| TOP74 | Z1_Cool_Curve_Outside_High_Temp | 30 | °C|
| TOP75 | Z1_Cool_Curve_Outside_Low_Temp | 20 | °C|
| TOP76 | Heating_Mode | 0 | Comp. Curve|
| TOP77 | Heating_Off_Outdoor_Temp | 17 | °C|
| TOP78 | Heater_On_Outdoor_Temp | -15 | °C|
| TOP79 | Heat_To_Cool_Temp | 15 | °C|
| TOP80 | Cool_To_Heat_Temp | 10 | °C|
| TOP81 | Cooling_Mode | 1 | Direct|
| TOP82 | Z2_Heat_Curve_Target_High_Temp | 35 | °C|
| TOP83 | Z2_Heat_Curve_Target_Low_Temp | 27 | °C|
| TOP84 | Z2_Heat_Curve_Outside_High_Temp | 15 | °C|
| TOP85 | Z2_Heat_Curve_Outside_Low_Temp | -5 | °C|
| TOP86 | Z2_Cool_Curve_Target_High_Temp | 10 | °C|
| TOP87 | Z2_Cool_Curve_Target_Low_Temp | 10 | °C|
| TOP88 | Z2_Cool_Curve_Outside_High_Temp | 30 | °C|
| TOP89 | Z2_Cool_Curve_Outside_Low_Temp | 20 | °C|
| TOP90 | Room_Heater_Operations_Hours | 6 | hours|
| TOP91 | DHW_Heater_Operations_Hours | 60 | hours|
| TOP92 | Heat_Pump_Model | E2 CF 0C 74 09 12 D0 0D 95 05 | Model|
| TOP93 | Pump_Duty | 108 | Duty|
| TOP94 | Zones_State | 2 | Zone1 and zone2 active|
| TOP95 | Max_Pump_Duty | 131 | Duty|
| TOP96 | Heater_Delay_Time | -1 | Minutes|
| TOP97 | Heater_Start_Delta | -128 | K|
| TOP98 | Heater_S| TOP_Delta | -128 | K|
| TOP99 | Buffer_Installed | 1 | Enabled|
| TOP100 | DHW_Installed | 1 | Enabled|
| TOP101 | Solar_Mode | 0 | Disabled|
| TOP102 | Solar_On_Delta | 10 | K|
| TOP103 | Solar_Off_Delta | 5 | K|
| TOP104 | Solar_Frost_Protection | 5 | °C|
| TOP105 | Solar_High_Limit | 80 | °C|
| TOP106 | Pump_Flowrate_Mode | -1 | unknown|
| TOP107 | Liquid_Type | 0 | Water|
| TOP108 | Alt_External_Sensor | 0 | Disabled|
| TOP109 | Anti_Freeze_Mode | 1 | Enabled|
| TOP110 | Optional_PCB | 1 | Enabled|
| TOP111 | Z1_Sensor_Settings | 1 | External Thermostat|
| TOP112 | Z2_Sensor_Settings | 1 | External Thermostat|
| TOP113 | Buffer_Tank_Delta | 10 | K|
| TOP114 | External_Pad_Heater | 0 | Disabled|
| TOP115 | Water_Pressure | -0.02 | Bar|
| TOP116 | Second_Inlet_Temp | -128 | °C|
| TOP117 | Economizer_Outlet_Temp | -128 | °C|
| TOP118 | Second_Room_Thermostat_Temp | -128 | °C|
| TOP119 | External_Control | 0 | Disabled|
| TOP120 | External_Heat_Cool_Control | 0 | Disabled|
| TOP121 | External_Error_Signal | 0 | Disabled|
| TOP122 | External_Compressor_Control | 0 | Disabled|
| TOP123 | Z2_Pump_State | 1 | On|
| TOP124 | Z1_Pump_State | 1 | On|
| TOP125 | TwoWay_Valve_State | 1 | Heat|
| TOP126 | ThreeWay_Valve_State2 | 0 | Room|
| TOP127 | Z1_Valve_PID | 100 | %|
| TOP128 | Z2_Valve_PID | 98.5 | %|
| TOP129 | Bivalent_Control | 0 | Disabled|
| TOP130 | Bivalent_Mode | 0 | Alternative|
| TOP131 | Bivalent_Start_Temp | -5 | °C|
| TOP132 | Bivalent_Advanced_Heat | 0 | Disabled|
| TOP133 | Bivalent_Advanced_DHW | 0 | Disabled|
| TOP134 | Bivalent_Advanced_Start_Temp | -8 | °C|
| TOP135 | Bivalent_Advanced_S| TOP_Temp | -2 | °C|
| TOP136 | Bivalent_Advanced_Start_Delay | 30 | Minutes|
| TOP137 | Bivalent_Advanced_S| TOP_Delay | 30 | Minutes|
| TOP138 | Bivalent_Advanced_DHW_Delay | 30 | Minutes|
| TOP139 | EEV_Valve | 120 | count|
| TOP140 | ByPass_Valve | 120 | count


Buffer Tank On Delta10, Z2 Dominant

| Topic | Name | Value | Description|
| --- | --- | --- | --- |
| TOP0 | Heatpump_State | 1 | On|
| TOP1 | Pump_Flow | 34.86 | l/min|
| TOP2 | Force_DHW_State | 0 | Disabled|
| TOP3 | Quiet_Mode_Schedule | 0 | Disabled|
| TOP4 | Operating_Mode_State | 4 | Heat+DHW|
| TOP5 | Main_Inlet_Temp | 26.75 | °C|
| TOP6 | Main_Outlet_Temp | 26.75 | °C|
| TOP7 | Main_Target_Temp | 48 | °C|
| TOP8 | Compressor_Freq | 0 | Hz|
| TOP9 | DHW_Target_Temp | 50 | °C|
| TOP10 | DHW_Temp | 49 | °C|
| TOP11 | Operations_Hours | 19518 | hours|
| TOP12 | Operations_Counter | 20615 | count|
| TOP13 | Main_Schedule_State | 0 | Disabled|
| TOP14 | Outside_Temp | 12 | °C|
| TOP15 | Heat_Power_Production | 0 | Watt|
| TOP16 | Heat_Power_Consumption | 0 | Watt|
| TOP17 | Powerful_Mode_Time | 0 | Off|
| TOP18 | Quiet_Mode_Level | 0 | Off|
| TOP19 | Holiday_Mode_State | 0 | Off|
| TOP20 | ThreeWay_Valve_State | 0 | Room|
| TOP21 | Outside_Pipe_Temp | 12 | °C|
| TOP22 | DHW_Heat_Delta | -10 | K|
| TOP23 | Heat_Delta | 5 | K|
| TOP24 | Cool_Delta | 5 | K|
| TOP25 | DHW_Holiday_Shift_Temp | -15 | K|
| TOP26 | Defrosting_State | 0 | Disabled|
| TOP27 | Z1_Heat_Request_Temp | 0 | °C|
| TOP28 | Z1_Cool_Request_Temp | 0 | °C|
| TOP29 | Z1_Heat_Curve_Target_High_Temp | 35 | °C|
| TOP30 | Z1_Heat_Curve_Target_Low_Temp | 27 | °C|
| TOP31 | Z1_Heat_Curve_Outside_High_Temp | 15 | °C|
| TOP32 | Z1_Heat_Curve_Outside_Low_Temp | -5 | °C|
| TOP33 | Room_Thermostat_Temp | 28 | °C|
| TOP34 | Z2_Heat_Request_Temp | 0 | °C|
| TOP35 | Z2_Cool_Request_Temp | 0 | °C|
| TOP36 | Z1_Water_Temp | 27 | °C|
| TOP37 | Z2_Water_Temp | 27 | °C|
| TOP38 | Cool_Power_Production | 0 | Watt|
| TOP39 | Cool_Power_Consumption | 0 | Watt|
| TOP40 | DHW_Power_Production | 0 | Watt|
| TOP41 | DHW_Power_Consumption | 0 | Watt|
| TOP42 | Z1_Water_Target_Temp | 28 | °C|
| TOP43 | Z2_Water_Target_Temp | 36 | °C|
| TOP44 | Error | No error | Error|
| TOP45 | Room_Holiday_Shift_Temp | -15 | K|
| TOP46 | Buffer_Temp | 27 | °C|
| TOP47 | Solar_Temp | -78 | °C|
| TOP48 | Pool_Temp | -78 | °C|
| TOP49 | Main_Hex_Outlet_Temp | 26 | °C|
| TOP50 | Discharge_Temp | 29 | °C|
| TOP51 | Inside_Pipe_Temp | 28 | °C|
| TOP52 | Defrost_Temp | 12 | °C|
| TOP53 | Eva_Outlet_Temp | 15 | °C|
| TOP54 | Bypass_Outlet_Temp | 16 | °C|
| TOP55 | Ipm_Temp | 17 | °C|
| TOP56 | Z1_Temp | 27 | °C|
| TOP57 | Z2_Temp | 27 | °C|
| TOP58 | DHW_Heater_State | 1 | Free|
| TOP59 | Room_Heater_State | 0 | Blocked|
| TOP60 | Internal_Heater_State | 0 | Inactive|
| TOP61 | External_Heater_State | 0 | Inactive|
| TOP62 | Fan1_Motor_Speed | 0 | r/min|
| TOP63 | Fan2_Motor_Speed | 0 | r/min|
| TOP64 | High_Pressure | 11 | Kgf/cm2|
| TOP65 | Pump_Speed | 2700 | r/min|
| TOP66 | Low_Pressure | 0 | Kgf/cm2|
| TOP67 | Compressor_Current | 0 | Ampere|
| TOP68 | Force_Heater_State | 0 | Inactive|
| TOP69 | Sterilization_State | 0 | Inactive|
| TOP70 | Sterilization_Temp | 62 | °C|
| TOP71 | Sterilization_Max_Time | 20 | Minutes|
| TOP72 | Z1_Cool_Curve_Target_High_Temp | 16 | °C|
| TOP73 | Z1_Cool_Curve_Target_Low_Temp | 10 | °C|
| TOP74 | Z1_Cool_Curve_Outside_High_Temp | 30 | °C|
| TOP75 | Z1_Cool_Curve_Outside_Low_Temp | 20 | °C|
| TOP76 | Heating_Mode | 0 | Comp. Curve|
| TOP77 | Heating_Off_Outdoor_Temp | 17 | °C|
| TOP78 | Heater_On_Outdoor_Temp | -15 | °C|
| TOP79 | Heat_To_Cool_Temp | 15 | °C|
| TOP80 | Cool_To_Heat_Temp | 10 | °C|
| TOP81 | Cooling_Mode | 1 | Direct|
| TOP82 | Z2_Heat_Curve_Target_High_Temp | 45 | °C|
| TOP83 | Z2_Heat_Curve_Target_Low_Temp | 35 | °C|
| TOP84 | Z2_Heat_Curve_Outside_High_Temp | 15 | °C|
| TOP85 | Z2_Heat_Curve_Outside_Low_Temp | -5 | °C|
| TOP86 | Z2_Cool_Curve_Target_High_Temp | 10 | °C|
| TOP87 | Z2_Cool_Curve_Target_Low_Temp | 10 | °C|
| TOP88 | Z2_Cool_Curve_Outside_High_Temp | 30 | °C|
| TOP89 | Z2_Cool_Curve_Outside_Low_Temp | 20 | °C|
| TOP90 | Room_Heater_Operations_Hours | 6 | hours|
| TOP91 | DHW_Heater_Operations_Hours | 60 | hours|
| TOP92 | Heat_Pump_Model | E2 CF 0C 74 09 12 D0 0D 95 05 | Model|
| TOP93 | Pump_Duty | 108 | Duty|
| TOP94 | Zones_State | 2 | Zone1 and zone2 active|
| TOP95 | Max_Pump_Duty | 131 | Duty|
| TOP96 | Heater_Delay_Time | -1 | Minutes|
| TOP97 | Heater_Start_Delta | -128 | K|
| TOP98 | Heater_S| TOP_Delta | -128 | K|
| TOP99 | Buffer_Installed | 1 | Enabled|
| TOP100 | DHW_Installed | 1 | Enabled|
| TOP101 | Solar_Mode | 0 | Disabled|
| TOP102 | Solar_On_Delta | 10 | K|
| TOP103 | Solar_Off_Delta | 5 | K|
| TOP104 | Solar_Frost_Protection | 5 | °C|
| TOP105 | Solar_High_Limit | 80 | °C|
| TOP106 | Pump_Flowrate_Mode | -1 | unknown|
| TOP107 | Liquid_Type | 0 | Water|
| TOP108 | Alt_External_Sensor | 0 | Disabled|
| TOP109 | Anti_Freeze_Mode | 1 | Enabled|
| TOP110 | Optional_PCB | 1 | Enabled|
| TOP111 | Z1_Sensor_Settings | 1 | External Thermostat|
| TOP112 | Z2_Sensor_Settings | 1 | External Thermostat|
| TOP113 | Buffer_Tank_Delta | 10 | K|
| TOP114 | External_Pad_Heater | 0 | Disabled|
| TOP115 | Water_Pressure | -0.02 | Bar|
| TOP116 | Second_Inlet_Temp | -128 | °C|
| TOP117 | Economizer_Outlet_Temp | -128 | °C|
| TOP118 | Second_Room_Thermostat_Temp | -128 | °C|
| TOP119 | External_Control | 0 | Disabled|
| TOP120 | External_Heat_Cool_Control | 0 | Disabled|
| TOP121 | External_Error_Signal | 0 | Disabled|
| TOP122 | External_Compressor_Control | 0 | Disabled|
| TOP123 | Z2_Pump_State | 1 | On|
| TOP124 | Z1_Pump_State | 1 | On|
| TOP125 | TwoWay_Valve_State | 1 | Heat|
| TOP126 | ThreeWay_Valve_State2 | 0 | Room|
| TOP127 | Z1_Valve_PID | 100 | %|
| TOP128 | Z2_Valve_PID | 100 | %|
| TOP129 | Bivalent_Control | 0 | Disabled|
| TOP130 | Bivalent_Mode | 0 | Alternative|
| TOP131 | Bivalent_Start_Temp | -5 | °C|
| TOP132 | Bivalent_Advanced_Heat | 0 | Disabled|
| TOP133 | Bivalent_Advanced_DHW | 0 | Disabled|
| TOP134 | Bivalent_Advanced_Start_Temp | -8 | °C|
| TOP135 | Bivalent_Advanced_S| TOP_Temp | -2 | °C|
| TOP136 | Bivalent_Advanced_Start_Delay | 30 | Minutes|
| TOP137 | Bivalent_Advanced_S| TOP_Delay | 30 | Minutes|
| TOP138 | Bivalent_Advanced_DHW_Delay | 30 | Minutes|
| TOP139 | EEV_Valve | 120 | count|
| TOP140 | ByPass_Valve | 120 | count|

**Comment by edterbak** [comment]
GREAT info. Thank you. I am digesting now.

Can you also say something about what happens when you SHIFT up or down?
With no buffer, z1 and compensation curve I can shift the curve up or down by sending SET16/SET5 command (depending on TOP111. 

The result of this shift, is visible in TOP29/TOP30. 
Does this remain the same when using a buffer? I suspect it does, but I need confirmation

**Comment by edterbak** [comment]
I have been looking at it

>So for example, z1 target temperature +35°, z2 target - +42°, and you keep bufger tank by 3k, so your main pump target temperature will be +45°. Z1 PID value will go from 100 to 0 by closing it's mixing valve to reduce it's flow temperature to ,+35°, and zone2 also in some time PID 2 will go down tu reduce it's supply temperature to ,42.. and in buffer tank it will try to keep temperature to 45, so it becames main target temperature by which it will reduce compressor frequency..

I do not see this in your pasted tables. The last table "Buffer Tank On Delta10, Z2 Dominant."
I see: 
z1 target = 28
z2 target = 36 (<< dominant)
Buffer Tank Delta = 10
Main target temp = 48

Main target temp should be 36+10=46. Right?

**Comment by geduxas** [comment]
> GREAT info. Thank you. I am digesting now.
> 
> Can you also say something about what happens when you SHIFT up or down? With no buffer, z1 and compensation curve I can shift the curve up or down by sending SET16/SET5 command (depending on TOP111.
> 
> The result of this shift, is visible in TOP29/TOP30 for zone1. I Think this is the same for zone 2 when it is dominant by looking at TOP82/TOP83. But I need this confirmed.
> 
> Additionally, does this behavior remain the same when using a buffer? I suspect it does, but I need confirmation

Yes shift is working in each zone independent. And dominant zone is which is highest target temperature, so by switching shifts, you are able to change dominant zone.. 

But appears that shift is not possible by enabling external thermostat.

**Comment by nofeardh** [comment]
Hi,

sorry for my late reply - I was away last few days.

I think Geduxas already described everything regarding Z1+Z2 behavior. Z2 is same story in your NodeRed workflow as Z1. Basicaly I think you just need to replicate code calculate WAR value and to make curve adjustments. Panasonic heatpump does everything else.

The other thing if you are using just Z2 emulation (like me) then you need to have some kind of logic to turn on/off water pump and to adjust mixing valve. I did this part in NodeRed.

I will describe installation in my house and maybe you will find something useful for you.

- two zones (Z1 - radiators, Z2 underfloor)
- using compensation curve for both zones
- I have parallel buffer installed (200L) but I do not have buffer option selected in Panasonic, because then my heat pump (generation H) behave like on-off heat pump.
- Z1 (radiators): water pump, NO mixing valve
- Z2 (underfloor): water pump + mixing valve
- using Heishamon controller + PCB emulation turned on
- with NodeRed I'm reading "panasonic_heat_pump/optional/Z2_Mixing_Valve" MQTT topic and if 1 then I'm closing mixing valve down for 10 sec, if 2 then I'm opening valve for 10 sec. 
- I'm reading panasonic_heat_pump/optional/Z2_Water_Pump MQTT value to operate water pumps....but this year I left water pump for underfloor allways on
- I have two DS18B20 sensors (connected to heishamon) to send current water temp for each zone every 2 minutes to Heishamon

So when using PCB emulation then there is some code needed to run water pump and mixing valves. But I don't think this should be  part of you flow....or maybe yes :)

**Comment by nofeardh** [comment]
@geduxas Yes as I said earlier I didn' know that pid info exists in Heishamon. I will check behaviour and change logic.

I have temp senzor for buffer otherwise there would be an error from pump. Maybe I will try with  buffer enabled once again next winter.

**Comment by edterbak** [comment]
> > GREAT info. Thank you. I am digesting now.
> > Can you also say something about what happens when you SHIFT up or down? With no buffer, z1 and compensation curve I can shift the curve up or down by sending SET16/SET5 command (depending on TOP111.
> > The result of this shift, is visible in TOP29/TOP30 for zone1. I Think this is the same for zone 2 when it is dominant by looking at TOP82/TOP83. But I need this confirmed.
> > Additionally, does this behavior remain the same when using a buffer? I suspect it does, but I need confirmation
> 
> Yes shift is working in each zone independent. And dominant zone is which is highest target temperature, so by switching shifts, you are able to change dominant zone..
> 
> But appears that shift is not possible by enabling external thermostat.

Yes, true. I remember. 
Shift is only possible in TOP111/112 = WATER option. 
In my current code for z1, I am not shifting by sending a value between -5 / +5. 
In my current code for z1 I am shifting by sending the entire new values for Z1_Heat_Curve_Target_High_Temp and Z1_Heat_Curve_Target_Low_Temp

Different topic. 
I still dont see the logic based on your tables @geduxas 

**Buffer Tank On Delta3, Z2 Dominant**
Main_Target_Temp	41
Z1_Water_Target_Temp	28
Z2_Water_Target_Temp	36
Buffer_Tank_Delta	3
Buffer_Temp	27

36 + 2 = 38... Computer says no
or alternatively 36 + 10 + 2 = 48... Computer says no...


**Buffer Tank On Delta10, Z1 Dominant**
Main_Target_Temp	48
Z1_Water_Target_Temp	**36**
Z2_Water_Target_Temp	28
Buffer_Tank_Delta	**10**
Buffer_Temp	27

36 + 10 = 46... Computer says no
or alternatively 36 + 10 + 2 = 48... Computer says .. yes. 
But is the alternative correct now??

Please help me with the numbers. 
It needs to be correct mathematically, and I need to anderstand and be sure it is correct. And currently I am still unsure and clueless.

**Comment by edterbak** [comment]
Can you test and see what the difference is in this calculation, when you set TOP111/TOP112 to WATER or the other thermostat types. 
Does the calculation stay the same, I would like to know.

**Comment by geduxas** [comment]
> Can you test and see what the difference is in this calculation, when you set TOP111/TOP112 to WATER or the other thermostat types. Does the calculation stay the same, I would like to know.

yes calculation stays same, 
here is snippet from service manual, maybe it will tell better logic:

![Image](https://github.com/user-attachments/assets/309cf20d-c279-4351-acd7-16a197ea4b94)

**Comment by edterbak** [comment]
Perfect! Thanks
